code/modules/crafting/recipes.dm

```diff
@@ -30,17 +30,25 @@
 	time = 15
 	category = CAT_WEAPON
 
-/datum/table_recipe/lance
-	name = "explosive lance"
+/datum/crafting_recipe/lance
+	name = "explosive lance (grenade)"
 	result = /obj/item/weapon/twohanded/spear
 	reqs = list(/obj/item/weapon/twohanded/spear = 1,
 				/obj/item/weapon/grenade = 1)
 	parts = list(/obj/item/weapon/grenade = 1)
 	time = 15
 	category = CAT_WEAPON
 
+/datum/crafting_recipe/lance2
+	name = "explosive lance (C4)"
+	result = /obj/item/weapon/twohanded/spear
+	reqs = list(/obj/item/weapon/twohanded/spear = 1,
+				/obj/item/weapon/c4 = 1)
+	parts = list(/obj/item/weapon/c4 = 1)
+	time = 40
+	category = CAT_WEAPON
 
```

Comment:
because it's redundant, since the first recipe covers C4 now. And the second one is trying to use a path that is inaccessible to players.